app/code/Magento/InventoryApi/Api/Data/SourceStockLinkInterface.php

```diff
@@ -0,0 +1,68 @@
+<?php
```

Comment:
we need to have dedicated Service interfaces (and implementations) which responsibility is to assign Sources to particular Stock

Something similar to:

Magento\Catalog\Api\CategoryLinkManagementInterface
Magento\Catalog\Api\CategoryLinkRepositoryInterface